R/endpoints.R

```diff
@@ -0,0 +1,46 @@
+api <- function() {
+  pr <- plumber$new()
+  pr$handle("POST", "/validate", handle_validate)
+  pr
+}
+
+handle_validate <- function(req, pjnz, shape, population, survey, programme,
+                            anc) {
+  ## TODO: Validate req json against the schema
+  valid <- validate_inputs(pjnz$path, shape$path, population$path, survey$path,
+                           programme$path, anc$path)
+  if (valid) {
+    response <- prepare_success_response(pjnz$path)
+  } else {
+    response <- prepare_fail_response()
+  }
+  ## TODO: Validate response against schema
+  response
+}
+
+prepare_success_response <- function(pjnz) {
+  country <- read_country(pjnz)
+  list(
+    "status" = "success",
+    "errors" = list(),
+    "data" = list(
+      "pjnz" = list(
+        "country" = country
+      ),
+      "shape" = list(),
+      "population" = list(),
+      "survey" = list(),
+      "programme" = list(),
+      "anc" = list()
+    )
+  )
+}
+
+prepare_fail_response <- function() {
+  list(
+    "status" = "failure",
+    "errors" = list(
+      "pjnz" = c("EXAMPLE_FAILURE")
```

Comment:
For errors, one thing that might be nice is if we have an enum of error types and return a simple list like `list(type = "InvalidInput", message = "The file was not the expected format")`.  We did that in montagu and when I was consuming the API it made it easier to tool around

Reply:
I was kind of envisioning a similar thing, something like

```
"errors" : {
  "pjnz": [
    "MISMATCH_COUNTRY_ERROR"
  ],
  "shape": [
    "MISMATCH_COUNTRY_ERROR",
    "MISMATCH_DISTRICTS_ERROR"
  ],
  "population": [
    "MISMATCH_DISTRICTS_ERROR",
  ],
...
```

Where the errors returned are some key which the front end can choose user readable wording for. Is there an advantage to returning a message along with that?